Week4/ReadingList/app/src/main/java/com/florencenjeri/readinglist/ui/BooksDetailsFragment.kt

```diff
@@ -0,0 +1,93 @@
+package com.florencenjeri.readinglist.ui
+
+import android.os.Bundle
+import android.util.Log
+import android.view.*
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.lifecycle.lifecycleScope
+import com.florencenjeri.readinglist.BooksDetailsFragmentArgs
+import com.florencenjeri.readinglist.R
+import com.florencenjeri.readinglist.ReadingListApplication
+import com.florencenjeri.readinglist.database.BooksDatabase
+import com.florencenjeri.readinglist.database.BooksRepository
+import com.florencenjeri.readinglist.model.Books
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.book_details_content.view.*
+import kotlinx.android.synthetic.main.books_details_fragment.*
+
+class BooksDetailsFragment : Fragment() {
+    private lateinit var booksViewModel: BooksViewModel
+    private lateinit var book: Books
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.books_details_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
+        val booksDao =
```

Comment:
You can just add this to a lazy delegate, and/or have the App class build the ViewModel.

Or use a Factory, that builds the ViewModel. https://medium.com/koderlabs/viewmodel-with-viewmodelprovider-factory-the-creator-of-viewmodel-8fabfec1aa4f